Fix failing CI on engine tests

```
=========================== short test summary info ============================
FAILED tests/ignite/engine/test_engine.py::TestEngine::test_run_check_triggered_events_on_iterator[False] - Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
 Emitted warnings: [].
FAILED tests/ignite/engine/test_engine.py::TestEngine::test_run_check_triggered_events_on_iterator[True] - Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
 Emitted warnings: [].
FAILED tests/ignite/engine/test_engine.py::TestEngine::test_distrib_gloo_cpu_or_gpu[False] - Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
 Emitted warnings: [].
FAILED tests/ignite/engine/test_engine.py::TestEngine::test_distrib_gloo_cpu_or_gpu[True] - Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted.
 Emitted warnings: [].
= 4 failed, 1792 passed, 528 skipped, 31 deselected, 14 warnings in 350.33s (0:05:50) =
```

--- a/tests/ignite/engine/test_engine.py
+++ b/tests/ignite/engine/test_engine.py
@@ -639,7 +639,7 @@ class TestEngine:
         self._test_check_triggered_events(limited_data_iterator(), max_epochs=1, epoch_length=100, exp_iter_stops=0)
         self._test_check_triggered_events(limited_data_iterator(), max_epochs=10, epoch_length=10, exp_iter_stops=0)
 
-        # These tests will fail
+        # These tests should fail
         with pytest.raises(AssertionError):
             with pytest.warns(UserWarning, match=r"Data iterator can not provide data anymore"):
                 self._test_check_triggered_events(limited_data_iterator(), max_epochs=3, epoch_length=100)
@@ -649,8 +649,10 @@ class TestEngine:
                 self._test_check_triggered_events(limited_data_iterator(), max_epochs=3, epoch_length=75)
 
         with pytest.raises(AssertionError):
-            with pytest.warns(UserWarning, match=r"Data iterator can not provide data anymore"):
-                self._test_check_triggered_events(limited_data_iterator(), max_epochs=1, epoch_length=101)
+            # Below test does not raise "Data iterator can not provide data anymore" warning as the last
+            # epoch is equal max_epochs
+            # with pytest.warns(UserWarning, match=r"Data iterator can not provide data anymore"):
+            self._test_check_triggered_events(limited_data_iterator(), max_epochs=1, epoch_length=101)
 
     def test_run_check_triggered_events_on_iterator(self):
         self._test_run_check_triggered_events_on_iterator()

